lib/classifier-reborn/bayes.rb

```diff
@@ -19,24 +21,27 @@ class Bayes
     #   threshold:        0.0    Default threshold, only used when enabled
     #   enable_stemmer:   true   When false, disables word stemming
     def initialize(*args)
-      @categories = {}
+      initial_categories = []
       options = { language:         'en',
                   auto_categorize:  false,
                   enable_threshold: false,
                   threshold:        0.0,
-                  enable_stemmer:   true
+                  enable_stemmer:   true,
+                  redis_con_str:    ''
```

Comment:
I liked this approach, thanks! Initially, I thought about letting users directly create instance of the Redis class and pass that as for connection, but that would be too much abstraction. I will go ahead and make the Memory backend as default and let the user pass an instance of the backend to allow override.